invenio_oauthclient/utils.py

```diff
@@ -163,15 +176,21 @@ def oauth_register(form, user_info=None, precedence_mask=None):
     :returns: A :class:`invenio_accounts.models.User` instance.
     """
     if form.validate():
-        data = form.to_dict()
+        form_data = form.data
 
         # let relevant information from the OAuth service's user info
         # have precedence over the values specified by the user
         if user_info:
             default_mask = {"email": True}
+            # act on form data so the `profile` is updated correctly
             filter_user_info(user_info, precedence_mask or default_mask)
-            patch_dictionary(data, user_info)
+            patch_dictionary(form_data, user_info)
 
+        # re-populate the form after appyling the precedence mask and
+        # convert the form data to user model's data
+        data = fill_form(form, form_data).to_dict()
+        # TODO: check why None values are there in the first place
```

Comment:
yeah, could we maybe create a ticket ?

Reply:
https://github.com/inveniosoftware/invenio-oauthclient/issues/275